rootfs/api/models/appsettings.py

```diff
@@ -86,6 +90,46 @@ def update_whitelist(self, previous_settings):
                     self.summary += ' and '
                 self.summary += "{} {}".format(self.owner, changes)
 
+    def update_autoscale(self, previous_settings):
+        data = getattr(previous_settings, 'autoscale', {}).copy()
+        new = getattr(self, 'autoscale', {}).copy()
+        # If no previous settings then do nothing
+        if not previous_settings:
+            return
+
+        # if nothing changed copy the settings from previous
+        if not new and data:
+            setattr(self, 'autoscale', data)
+        elif data != new:
+            for proc, scale in new.items():
+                if scale is None:
+                    # error if unsetting non-existing key
+                    if proc not in data:
+                        raise UnprocessableEntity('{} does not exist under {}'.format(proc, 'autoscale'))  # noqa
+                    del data[proc]
+                else:
+                    data[proc] = scale
```

Comment:
Inclined to say we solve that problem when we get to it unless we have a clear cut way towards that right now. @slack ?